src/coreclr/pal/inc/rt/sal.h

```diff
@@ -2649,10 +2649,10 @@ buffer, use the table in the buffer annotations section.
 */
 
 // These macros conflict with c++ headers.
-#ifndef PAL_STDCPP_COMPAT
+#ifdef PAL_NO_STDCPP_COMPAT
```

Comment:
Why was this flipped? I would expect that we will never define or undefine `__in` when PAL_NO_STDCPP_COMPAT is defined.

Reply:
Technically these lines are no longer necessary since these macros are no longer used in the code.

I tried to define PAL_STDCPP_COMPAT globally but this has much wider change to the code and nothing compiles.

In order to compile and have an ability to define macros for some rare case if ever exists I flipped the condition.

Reply:
I do not think we need to worry about some potential future rare cases. I think this condition should either stay as is, or it should be deleted.

Reply:
The condition cannot be the same cause this is the problem PR solves ;)
I can remove it completely.
Will push the change soon.